### PR TITLE
Improve/input validation

### PR DESCRIPTION
**Pull request by zenangst**
The idea was to decouple the type of field and the input validation.

Phone numbers need to be stored as strings (leading zeros being the issue) but should only contain number. The field is set to `text` but you can override this type by specifying `input` in the validation object.

As an added bonus, the `input` also configures the keyboard.

### CODE REVIEW

**Comment by NSElvis** [comment]
I think supporting
```
"type":"phone_numer"
```

Might be a better idea.

https://github.com/hyperoslo/Form/issues/202

**Comment by zenangst** [comment]
What about postal codes and bank account numbers?

**Comment by NSElvis** [comment]
`phone_number` and `email` are generic enough to have `native` support, for everything else you'll use 

```
"validations":{
                "format":"[\\w._%+-]+@[\\w.-]+\\.\\w{2,}",
                "required":true
              }
```

**Comment by zenangst** [comment]
How would this help to specify which keyboard should be used?

Let's say that you have bank account number field:

```JSON
{
  "id":"bank_account_number",
  "type":"text",
  "size":{
    "width":35,
    "height":1
},
  "validations":{
    "input":"0123456789",
    "required":true,
    "min_length":11,
    "max_length":11
  }
}
```

Here you would like it to store it as text (to support leading zeros), so you set the `type` to be `text`.
You then use the `input` key to add allowed characters. What we don't get in this scenario is the right keyboard layout.

If we change the `type` of the field to be `number`, then the keyboard layout is correct but you end up with a different `rawFieldValue` and you lose the ability to store leading zeros.

Am I missing something? :)

**Comment by NSElvis** [comment]
Oh, I see! What about supporting something like:

{
  "id":"bank_account_number",
  "type":"text",
  "input_type":"numeric",
  "size":{
    "width":35,
    "height":1
}

I don't like the keyboard type inside the validations since it's not really a validation.

**Comment by zenangst** [comment]
@NSElvis Yah that makes more sense, I put it inside the validations because it would by proxy only allow the right set of characters but this makes more sense. Let me refactor the PR a bit!

**Comment by zenangst** [comment]
@NSElvis what do you think of this?

File: Source/Models/FORMField.m [hunk 1]
```diff
@@ -35,6 +35,12 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
     _subtitle = [dictionary andy_valueForKey:@"subtitle"];
     _typeString  = [dictionary andy_valueForKey:@"type"];
     _type = [self typeFromTypeString:self.typeString];
+    _inputType = [dictionary andy_valueForKey:@"input_type"];
+
+    if (_inputType) {
+        _typeString = _inputType;
```
Comment (NSElvis) [inline Source/Models/FORMField.m]: This part is odd, can you make it in a way that you don't overwrite the typeString? It's coupled with other logic, meanwhile the inputType is only used for the keyboard

**Comment by NSElvis** [comment]
@RamonGilabert thanks!

**Pull request merged**